FWCore/ParameterSet/interface/ParameterSetDescription.h

```diff
@@ -149,48 +171,65 @@ namespace edm {
-      return addVPSet<U>(iLabel, validator, defaults, true, false, true);
+      return addVPSet<U>(iLabel, validator, defaults, true, Modifier::kNone, true);
     }
 
     template <typename U>
     ParameterDescriptionBase* addVPSetUntracked(U const& iLabel,
                                                 ParameterSetDescription const& validator,
                                                 std::vector<ParameterSet> const& defaults) {
-      return addVPSet<U>(iLabel, validator, defaults, false, false, true);
+      return addVPSet<U>(iLabel, validator, defaults, false, Modifier::kNone, true);
     }
 
     template <typename U>
     ParameterDescriptionBase* addVPSetOptional(U const& iLabel,
                                                ParameterSetDescription const& validator,
                                                std::vector<ParameterSet> const& defaults) {
-      return addVPSet<U>(iLabel, validator, defaults, true, true, true);
+      return addVPSet<U>(iLabel, validator, defaults, true, Modifier::kOptional, true);
     }
 
     template <typename U>
     ParameterDescriptionBase* addVPSetOptionalUntracked(U const& iLabel,
                                                         ParameterSetDescription const& validator,
                                                         std::vector<ParameterSet> const& defaults) {
-      return addVPSet<U>(iLabel, validator, defaults, false, true, true);
+      return addVPSet<U>(iLabel, validator, defaults, false, Modifier::kOptional, true);
     }
 
     template <typename U>
     ParameterDescriptionBase* addVPSet(U const& iLabel, ParameterSetDescription const& validator) {
-      return addVPSet<U>(iLabel, validator, true, false, true);
+      return addVPSet<U>(iLabel, validator, true, Modifier::kNone, true);
     }
 
     template <typename U>
     ParameterDescriptionBase* addVPSetUntracked(U const& iLabel, ParameterSetDescription const& validator) {
-      return addVPSet<U>(iLabel, validator, false, false, true);
+      return addVPSet<U>(iLabel, validator, false, Modifier::kNone, true);
     }
 
     template <typename U>
     ParameterDescriptionBase* addVPSetOptional(U const& iLabel, ParameterSetDescription const& validator) {
-      return addVPSet<U>(iLabel, validator, true, true, true);
+      return addVPSet<U>(iLabel, validator, true, Modifier::kOptional, true);
     }
 
     template <typename U>
     ParameterDescriptionBase* addVPSetOptionalUntracked(U const& iLabel, ParameterSetDescription const& validator) {
-      return addVPSet<U>(iLabel, validator, false, true, true);
+      return addVPSet<U>(iLabel, validator, false, Modifier::kOptional, true);
+    }
+
+    template <typename U>
+    ParameterDescriptionBase* addVPSetOptionalUntracked(U const& iLabel) {
+      ParameterSetDescription validator;
+      return addVPSet<U>(iLabel, validator, false, Modifier::kObsolete, true);
```

Comment:
(as we discussed) This overload looks wrong
* it allows adding optional VPSet without a validator
* the modifier is `Obsolete`

Maybe this overload just needs to be removed?

Reply:
removed